### DISCUSSION TITLE
Question: why Ildaeil instead of Carla?

### DISCUSSION THREAD

**Discussion by SanderVocke** [DISTRHO/Ildaeil#21] [Q&A] [answered]
I hope this is a suitable place for me to ask a question. I stumbled upon this project in the KXStudio news page. I was intrigued right away because of the stated goal: providing a minimal wrapper plugin to host others in.

It just so happens that in my project [ShoopDaLoop](https://github.com/SanderVocke/shoopdaloop), I had been using Carla in this way: ShoopDaLoop has a minimal LV2 host implementation that can only load Carla via LV2, with the idea that any other plugins can then be loaded in Carla from there. This sounds like exactly what Ildaeil is meant for, so I might consider switching.

After reading the front page a bit, I am wondering though: what does Idaeil offer compared to Carla itself? Carla can already be loaded as a plugin in various formats to wrap other plugins, so why would you need a(-nother) wrapper around it?

Cheers!

**Comment by falkTX** [comment, marked as answer]
Carla as plugin has its own GUI to manage things, acting as a UI layer between the loaded plugin(s) inside carla and carla as plugin.
For some cases we do not want that, we want to simply load plugin X in host Y without bothering with a layer in between.
This is why Ildaeil will embed the loaded plugin UI by default, thus making the experience as seamless as possible. There is still a layer in between things but it is much smaller and easier to deal with.
Also Ildaeil is meant to load 1 plugin at a time, purposefully.

And yeah, all that Ildaeil does Carla does too, because it is Carla doing all the heavy work behind the scenes. But Carla as a project has a different goal

**Reply by SanderVocke** [reply]
That makes sense, thanks for the quick reply! 

I might add support for both, so users can pick their preference.
